### PR TITLE
[4.0] Module description not a manual

### PR DESCRIPTION
**Pull request by brianteeman**
This is not the place to write a manual or help document

### CODE REVIEW

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 5167a159de0bea846c653d7a164bbda25a472850<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20695">issues.joomla.org/tracker/joomla-cms/20695</a>.</sub>

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 5167a159de0bea846c653d7a164bbda25a472850<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20695">issues.joomla.org/tracker/joomla-cms/20695</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20695">issues.joomla.org/tracker/joomla-cms/20695</a>.</sub>

File: language/en-GB/en-GB.mod_languages.sys.ini [hunk 1]
```diff
@@ -4,6 +4,5 @@
...
-MOD_LANGUAGES_LAYOUT_DEFAULT="Default"
-
+MOD_LANGUAGES_XML_DESCRIPTION="This module displays a language switcher on your website of available content languages."
```
Comment (Quy) [inline language/en-GB/en-GB.mod_languages.sys.ini]: Text not the same.

Comment (brianteeman) [reply language/en-GB/en-GB.mod_languages.sys.ini]: not the same as what?

Comment (Quy) [reply language/en-GB/en-GB.mod_languages.sys.ini]: The 2 strings are not identical.

> This module displays a language switcher of available content languages on your website.
> This module displays a language switcher on your website of available content languages.

Comment (brianteeman) [reply language/en-GB/en-GB.mod_languages.sys.ini]: ah sorry - i changed one - fixing now

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on d015860676b0fd1144db049b8f1bc66004d8b3dd<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20695">issues.joomla.org/tracker/joomla-cms/20695</a>.</sub>

**Comment by infograf768** [comment]
RANT 👎 . 
Taking off instructions of use when they already exist in all our language strings makes no sense. 
It just a fake simplification!!!
**It looks like some people just want newbies to choose another CMS.**

That does not mean these tips should not be modified in 4.0 as displaying the language field now depends on the fact that the language filter is enabled or not. Also, normally, the multilanguage status module is now automatically published.

As repeated multiple times (and never listened to...) it is useless to looks in the docs for full documentation as most are missing in a user specific language and would only be present in en-GB.

These tips exist for a good reason and taking them off ONLY pleases the English-speaking who are not concerned by users who are not English-fluent or joomla elders who already know what this module does (in the case of this module).

As usual, maintainers will do whatever they want and they always unhappily just merge that type of PR.

**Comment by brianteeman** [comment]
1. It's not a simplification it's a correction. The more you write the less people read. That's a scientific fact.
2. The text itself, if you read it, is garbage. It's both bad English and more confusing than helpful.
3. It's called a description not a manual
4. Years ago the docs site wasn't translatable. It is now and has been for many years.
5. There is nothing to stop any TT from writing whatever they want. It does need t have to be a literal translation. We have already seen that many times.

**Comment by dgrammatiko** [comment]
> It looks like some people just want newbies to choose another CMS.

Let me rephrase that as you got it wrong: `It looks like some people want only the guys from 1.5 era to use this CMS.`

**Comment by infograf768** [comment]
I will get out this PR as it gets on my nerves.
Just one last word

> Years ago the docs site wasn't translatable. It is now and has been for many years.

That is pure bullshit and any non-english speaker knows the reality of docs translations.
That was all. Do whatever you want.

**Comment by alikon** [comment]
too much of all for this, folks we have bigger issues to spend time in

**Comment by ot2sen** [comment]
A concise and descriptive text is optimal for the description. Especially for people new to the CMS.

The useful longer help text from before that is now replaced should be added to the corresponding help page https://docs.joomla.org/Help38:Extensions_Module_Manager_Language_Switcher

That way people won´t get overwhelmed if they are new and they still have the 'Help' button in reach if they need assistance understanding how this module works.

**Comment by zwiastunsw** [comment]
My opinion as a translator: I understand @infograf768  intentions. But @brianteeman is right in that case. It must be a description, not an instruction. Let it stay that way in Joomla3, but in Joomla 4 it is necessary to do it, as proposed by Brian. 
The same is true for the 'description' of the 'Alias' field label. Instead of a description, we have a school essay.

**Comment by dgrammatiko** [comment]
Well actually there should be zero-null-none descriptions.
- Why?
- Because quite frankly J4 will come with a guided tour per page: https://github.com/joomla-projects/gsoc17_helpscreens_on_jdocs/pull/7

We're trying very hard to keep old fashioned crappy designs when people already have spend a great amount of time to actually move the codebase forward. So please remove **all** the descriptions, tooltips and popups from the form and merge that guided tour so you can add those essays kinda descriptions there.

**Comment by alikon** [comment]
most of the times if not all the times , the gsoc projects never reach the end user, this is a very very bad thing

**Comment by zwiastunsw** [comment]
One more thought on this occasion. Many descriptions make no sense at all.  A significant label and significant buttons will suffice for a description. For example, wherever you have the Show/Hide or Yes/No buttons.  I therefore agree with @dgrammatiko.

**Comment by dgrammatiko** [comment]
@alikon that GSOC project was a pet project to move the data to a json file, the original code was already done here: https://github.com/joomla/40-backend-template/pull/247/files

Also @brianteeman you were part of the London sprint where we extensively discussed about dropping the descriptions (due to the fact that J4 will come with a guided tour) and actually you were the one that did the heavy lifting of removing the majority of descriptions. So let's stick to the original plan of removing all the descriptions (maybe altering some labels to be more specific) and merging the code from that GSOC project. Let's move forward instead of restyling the bad aspects of J3 design...

**Comment by brianteeman** [comment]
This is not a tooltip please don't confuse it as one. Almost all of those were removed. This is the description of the module itself which is displayed on screen and not as a tooltip/popover.

I wish people read the pr before ranting

**Comment by dgrammatiko** [comment]
It seems like a tooltip to me: 
screen shot 2018-06-09 at 14 07 21

**Comment by brianteeman** [comment]
Didn't even realise it was shown there. Tooltips there should be killed as they serve no purpose. I was only looking at the display in the module itself when you open it and it's so long it gets truncated and split into a separate tab

**Comment by brianteeman** [comment]
alhough its a seperate issue the display of tooltips on that screen is just plain wrong in both j3 and j4. Why on earth would you hover over a sentence to read the same sentence

**Comment by mbabker** [comment]
@dgrammatiko That's a case where the string was reused in another context.  The extension descriptions are generally only shown on install/update and in module and plugin edit pages for those extension types and that's where their use should stay.  If we insist on having tooltip descriptions on the modules list or the manage extensions page (and IMO core should not make it 100% impossible to use tooltips or whatever when the use case calls for something extra), there should be a dedicated more succinct string used versus that one, especially as extension developers frequently use HTML in that description string and create rather lengthy descriptions with full fledged styling.

screen shot 2018-06-09 at 7 27 48 am

**Comment by dgrammatiko** [comment]
Ahaha, does that tooltip actually contains links? Sooooo wrong, WAI-ARIA best practices denote that there should be 0 interaction with the content of the tooltip (only reading the content)...
YCFS...

**Comment by mbabker** [comment]
Not their fault.  As I said, that string is used for new install and updates, when it shows in that context it's all OK.  Since we reuse that same string as a tooltip, that's what creates problems.

**Comment by dgrammatiko** [comment]
So let's have a short and long description, so both cases are covered (PWA manifests have a "name" and a "short_name" so I guess we can apply the same concept here as well: https://developer.mozilla.org/en-US/docs/Web/Manifest )

**Comment by brianteeman** [comment]
The module screen is different as the text is a tooltip when you hover over the same text. Either way it is off topic for this pr

**Comment by brianteeman** [comment]
Thanks

**Comment by infograf768** [comment]
I would have rather done, but I understand that whoever is always right.
` +MOD_LANGUAGES_XML_DESCRIPTION="This module lets you switch between available content languages on your website when the site is set as multilingual."`

**Comment by brianteeman** [comment]
Maybe you shouldn't have thrown your toys out of the pram and then this could have been updated with your suggestion. As it has now been merged you will have to submit a pull request.
